src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp

```diff
@@ -104,66 +120,109 @@ ExitCode PlatformInconsistencyCheckerWorker::checkTree(std::shared_ptr<Node> rem
+                setPauseDone();
+            }
+
+            Utility::msleep(LOOP_PAUSE_SLEEP_PERIOD);
+        }
+
+        ExitCode exitCode = checkLocalTree(childeIt->second, parentPath / localNode->name());
+        if (exitCode != ExitCode::Ok) {
+            return exitCode;
+        }
+    }
+    return ExitCode::Ok;
+}
+
+void PlatformInconsistencyCheckerWorker::blacklistNode(const std::shared_ptr<Node> node,
                                                        const InconsistencyType inconsistencyType) {
     // Local node needs to be excluded before call to blacklistTemporarily because
     // we need the DB entry to retrieve the corresponding node
     NodeIdPair nodeIDs;
+    const std::shared_ptr<Node> localNode = node->side() == ReplicaSide::Local ? node : correspondingNodeDirect(node);
+    const std::shared_ptr<Node> remoteNode = node->side() == ReplicaSide::Remote ? node : correspondingNodeDirect(node);
 
-    if (const auto localNode = correspondingNodeDirect(remoteNode); localNode) {
-        // Also exclude local node by adding "conflict" suffix
+    if (localNode) {
         const SyncPath absoluteLocalPath = _syncPal->localPath() / localNode->getPath();
         LOGW_SYNCPAL_INFO(_logger,
-                          L"Excluding also local item with " << Utility::formatSyncPath(absoluteLocalPath).c_str() << L".");
-        PlatformInconsistencyCheckerUtility::renameLocalFile(absoluteLocalPath,
-                                                             PlatformInconsistencyCheckerUtility::SuffixTypeConflict);
+                          L"Excluding local item with " << Utility::formatSyncPath(absoluteLocalPath).c_str() << L".");
+        PlatformInconsistencyCheckerUtility::renameLocalFile(
+                absoluteLocalPath, node->side() == ReplicaSide::Remote
+                                           ? PlatformInconsistencyCheckerUtility::SuffixType::Conflict
+                                           : PlatformInconsistencyCheckerUtility::SuffixType::Blacklisted);
+
+        if (node->side() == ReplicaSide::Local) {
+            bool found = false;
+            DbNodeId dbId = -1;
+            if (!_syncPal->syncDb()->dbId(ReplicaSide::Local, *localNode->id(), dbId, found)) {
+                LOGW_WARN(_logger,
+                          L"Failed to retreive dbId for local node: " << Utility::formatSyncPath(absoluteLocalPath).c_str());
+            }
+            if (found && !_syncPal->syncDb()->deleteNode(dbId, found)) {
+                // Remove node (and childs by cascade) from DB if it exists (else ignore as it is already not in DB)
+                LOGW_WARN(_logger, L"Failed to delete node from DB: " << Utility::formatSyncPath(absoluteLocalPath).c_str());
+            }
 
-        if (localNode->id().has_value()) nodeIDs.localId = *localNode->id();
+            if (!_syncPal->vfsFileStatusChanged(absoluteLocalPath, node->side() == ReplicaSide::Remote ? SyncFileStatus::Conflict
```

Comment:
A small helper would be nice, mostly to document what is the the expected outcome and why.

Reply:
[c020720](https://github.com/Infomaniak/desktop-kDrive/pull/347/commits/c020720b986aaa8c951dc2bc2d6af8629d442000)